dbt/__init__.py

```diff
@@ -0,0 +1,3 @@
+from pkgutil import extend_path
+
+__path__ = extend_path(__path__, __name__)
```

Comment:
I've read changing __path__ this way is uncommon. What's the reason here? Super curious about the use case.

Reply:
We had a convo [in Slack](https://dbt-labs.slack.com/archives/G01GRE7RQSF/p1651772259078419) here about it but the TLDR is that we reuse `dbt` namespace twice which confuses mypy. This helps just combine them for mypy to reference. In the long run, we should refactor the codebase to not have this issue